contracts/stake-cw721/src/contract.rs

```diff
@@ -0,0 +1,1844 @@
+                    .clone()
+                    .unwrap_or_else(|| DEFAULT_ADDR.clone()),
+                amount_to_claim,
+                duration.after(&env.block),
+            )?;
+
+            Ok(Response::new()
+                .add_attribute("action", "unstake")
+                .add_submessages(hook_msgs)
+                .add_attribute("from", info.sender)
+                .add_attribute("token_id", token_id)
+                .add_attribute("claim_duration", format!("{}", duration))
+                .add_attribute(
+                    "reward_wallet",
+                    validated_reward_wallet.unwrap_or_else(|| DEFAULT_ADDR.clone()),
+                )
+                .add_attribute("reward_amount", amount_to_claim))
+        }
+    }
+}
+
+pub fn execute_claim_nfts(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+) -> Result<Response, ContractError> {
+    let nfts = NFT_CLAIMS.claim_nfts(deps.storage, &info.sender, &_env.block)?;
+    if nfts.is_empty() {
+        return Err(ContractError::NothingToClaim {});
+    }
+
+    let config = CONFIG.load(deps.storage)?;
+    let mut msgs = Vec::new();
+    for nft in nfts.iter() {
+        let cw_transfer_msg = cw721::Cw721ExecuteMsg::TransferNft {
+            recipient: info.sender.to_string(),
+            token_id: nft.clone(),
+        };
+
+        let wasm_msg = cosmwasm_std::WasmMsg::Execute {
+            contract_addr: config.nft_address.to_string(),
+            msg: to_binary(&cw_transfer_msg)?,
+            funds: vec![],
+        };
+        msgs.push(wasm_msg);
+    }
+
+    Ok(Response::new()
+        .add_messages(msgs)
+        .add_attribute("action", "claim_nfts")
+        .add_attribute("from", info.sender.clone()))
+}
+
+pub fn execute_claim_rewards(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+) -> Result<Response, ContractError> {
+    let reward_amount =
+        REWARD_CLAIMS.claim_tokens(deps.storage, &info.sender, &_env.block, None)?;
+    if reward_amount.is_zero() {
+        return Err(ContractError::NothingToClaim {});
+    }
+
+    let config = CONFIG.load(deps.storage)?;
+    let cw_send_msg = cw20::Cw20ExecuteMsg::Transfer {
+        recipient: info.sender.to_string(),
+        amount: reward_amount,
+    };
+
+    let wasm_msg = cosmwasm_std::WasmMsg::Execute {
+        contract_addr: config.reward_token_address.unwrap().to_string(),
+        msg: to_binary(&cw_send_msg)?,
+        funds: vec![],
+    };
+
+    Ok(Response::new()
+        .add_message(wasm_msg)
+        .add_attribute("action", "claim_reward")
+        .add_attribute("from", info.sender)
+        .add_attribute("amount", reward_amount))
+}
+
+pub fn execute_update_config(
+    info: MessageInfo,
+    deps: DepsMut,
+    new_owner: Option<String>,
+    new_manager: Option<String>,
+    duration: Option<Duration>,
+) -> Result<Response, ContractError> {
+    let new_owner = new_owner
+        .map(|new_owner| deps.api.addr_validate(&*new_owner))
```

Comment:
What is the `&*` doing here? Possible we really do need this and my Rust reading just isn't good enough.

Reply:
No, actually we do not need &*. Updated it!